recipe/meta.yaml

```diff
@@ -1,46 +1,105 @@
 {% set name = "cryptominisat" %}
 {% set version = "5.6.5" %}
 {% set sha256 = "b2751f8380a59c4885bea4c297536f0af2230306b1458d3e6b78d6e29f37b9d2" %}
+{% set build = 0 %}
+{% set m4ri_version = 20140914 %}
 
 package:
-  name: '{{ name|lower }}'
-  version: '{{ version }}'
+  name: cryptominisat-split
+  version: {{ version }}
 
 source:
-  fn: {{ version }}.tar.gz
-  url: https://github.com/msoos/{{ name }}/archive/{{ version }}.tar.gz
+  url: https://github.com/msoos/cryptominisat/archive/{{ version }}.tar.gz
   sha256: {{ sha256 }}
+  patches:
+    - remove-debug-symbols.patch
 
 build:
-  number: 0
-  skip: True  # [win]
-  run_exports:
-   - {{ pin_compatible('cryptominisat', max_pin='x.x') }}
+  number: {{ build }}
+  skip: True  # [win and vc<14]
 
 requirements:
   build:
     - {{ compiler('cxx') }}
     - cmake
   host:
-    - python
-    - setuptools
-    - boost-cpp 1.67.0
     - zlib
-    - m4ri 20140914
-  run:
-    - python
-    - boost-cpp 1.67.0
-    - zlib
-    - m4ri 20140914
 
-test:
-  imports:
-    - pycryptosat
-  commands:
-    - conda inspect linkages -p $PREFIX cryptominisat  # [not win]
-    - conda inspect objects -p $PREFIX cryptominisat  # [osx]
-    - echo -e '1 0\n-1 0' | cryptominisat5 | grep UNSATISFIABLE
-    - echo -e '1 0\n1 -1 0' | cryptominisat5 | grep SATISFIABLE
+outputs:
+  - name: libcryptominisat
+    build:
+      run_exports:
+        - {{ pin_subpackage('libcryptominisat', max_pin='x.x') }}
+    script: install-lib.sh
+    requirements:
+      build:
+        - {{ compiler('cxx') }}
+        - cmake
+        - make
+      host:
+        - zlib
+        - m4ri {{ m4ri_version }}  # [unix and gpl_ok]
+      run:
+        # gpl_ok stuff requires creation of a gpl metapackage.
+        # It should have a track_feature, so that it is unfavorable unless explicitly requested.
```

Comment:
I'm not terribly fond of putting a `track_features: [gpl]` on such a package... But I know it's likely the only solution we currently have to somewhat make `conda` prefer the `gpl_ok` variety when that `gpl` package is installed.
To avoid the `gpl_ok` build, it would be even fine to have a `gpl` package without `track_features` and just prevent its installation (via `pinned_packages: [ 'gpl <0' ]` or something since `disallowed_packges` still doesn't quite work well). But without that `track_features: [gpl]`, the solver won't, even if `gpl` is installed, prefer the build with more dependencies...

We currently just have no really reliable way to express preferences between build variants in `conda` :disappointed:..
(I currently don't have the brain capacity to think about, but at some point I'd like to revisit https://github.com/conda/conda/issues/6950, or a generalization of it, which might enable us to express some ordering between variants.)

Reply:
Yes, that's not the right way to do this.  I propose omitting the GPL functionality for now, but we really need to make that work.

Alternatively, we could make two outputs now that are mutually exclusive somehow, or that clobber each other (with similar files).  Perhaps something like libcms and libcms-gpl?

Terrible hack, but I'm not sure we really can do something better right now.

Reply:
> I propose omitting the GPL functionality for now

That is something we could do for `defaults`, yes. I'm not sure how the @conda-forge/cryptominisat maintainers feel about doing this on `conda-forge`, though. Arguably, this could be seen as some kind of regression for the package here. But if they think it would be fine, then that's fine by me, too.

> I'm not sure we really can do something better right now.

Well, theoretically... :smile: :

- Make `conda-forge` a multichannel like `defaults`, at least consisting of `conda-forge/label/main` and `conda-forge/label/alt` (or something).
- Create an empty `gpl` package (completely empty, not even `track_features`).
- Make the `gpl_ok=True`-variant depend on that `gpl` package and put it on `conda-forge/label/main` (since the `gpl_ok=True` one can be given preference here).
- Put the `gpl_ok=False` variant on `conda-forge/label/alt`.
- Success!
  - With that, we just made it possible to express preference between build variants!
  - On default, `conda install libcryptominisat` would install the GPL version with all its features.
  - If we want to prevent installation of the GPL variant,
    `conda config --add pinned_packages 'gpl <0' && conda install libcryptominisat` will give us the one from `conda-forge/label/alt`

Too bad that multichannel configuration is only possible on the user side....

Reply:
> How much effort do you think that might take?

I'm not totally sure what you mean. Do you mean how much effort it would be to make multichannels something that can be done server-side? Idk, you would need to add some special key to `repodata.json` (or similar) on `anaconda.org` and teach `conda` about it. Shouldn't be too hard to implement in `conda`, I suppose. But no idea in regards to `anaconda.org` or the effects on downstream projects that download/parse `repodata.json` on their own instead of through `conda`...

If you don't do it server-side, you'd have to tell the user to put `conda-forge/label/alt` into their channel list or create a `custom_multichannel` with that (for which they might run into https://github.com/conda/conda/issues/7022, though...).

If, on the other hand, you mean how to tell the CI to put those variant under different labels, *maybe* you could try to do something akin to https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/a7054f659eb1ee01a0e5dce29b931b7edf68e9a3/recipe/conda_build_config.yaml#L70-L72, but with
```yaml
  - conda-forge alt  # [not gpl_ok]
```
instead of
```
  - conda-forge main
```
. Telling the CI to use those channel would be done with
```yaml
  - conda-forge/label/alt,conda-forge/label/main,defaults
```
or something like in https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/a7054f659eb1ee01a0e5dce29b931b7edf68e9a3/recipe/conda_build_config.yaml#L66-L68.
Though, that's a little bit of speculating on my part since I don't know well all that works and it's certainly not something common to do...